docs/networking/vpn/secure-clustering-with-tinc-on-ubuntu-16-04.md

```diff
@@ -0,0 +1,102 @@
+---
+author:
+  name: Linode Community
+  email: docs@linode.com
+description: 'Use tinc VPN to securely cluster and access services on Ubuntu 16.04'
+keywords: 'tinc,vpn,networking,ubuntu'
+license: '[CC BY-ND 4.0](https://creativecommons.org/licenses/by-nd/4.0)'
+title: 'Secure Clustering with tinc on Ubuntu 16.04'
+contributor:
+  name: Sam Mauldin
+published: ''
+modified: Monday, September 12th, 2016
+modified_by:
+    name: Alex Fornuto
+---
+
+tinc is a meshed VPN that can be used to encrypt traffic between a cluster of Linodes. For example, if you're running a MySQL database, and need more performance, you can setup a cluster of nodes and secure data as it passes through the network.
+
+## Installing tinc
+
+1.  Make sure your systems are up to date.
+
+        sudo apt-get update
+        sudo apt-get upgrade
+
+2.  Then install the `tinc` package on the nodes you want to connect.
+
+        sudo apt-get install tinc
+
+## Configure tinc
+
+3.  Decide what you want to name your network and put it in `/etc/tinc/nets.boot`.
+
+        echo 'mynetwork' | sudo tee -a /etc/tinc/nets.boot
+
+4.  Create a directory to put the network configuration files in.
+
+        sudo mkdir /etc/tinc/mynetwork/
+
+5.  Make up and down scripts to be run by tinc.
+
+In the tinc-up file, you specify the IP address to be used by each node. In this example, `192.168.100.xxx` is used.
+
+{: .file }
+/etc/tinc/mynetwork/tinc-up
+:   ~~~ conf
+    #!/bin/bash
+    ifconfig $INTERFACE 192.168.100.1 netmask 255.255.255.0
+    ~~~
+
+{: .file }
+/etc/tinc/mynetwork/tinc-down
+:   ~~~ conf
+    #!/bin/bash
+    ifconfig $INTERFACE down
+    ~~~
+
+6.  Make sure the `tinc-up` and `tinc-down` files are executable.
+
+        sudo chmod +x /etc/tinc/mynetwork/tinc-*
+
+7.  Create a tinc.conf file. You'll need to use a seperate name for every node. The `ConnectTo` variable specifies which other nodes your server will connect to directly when it starts.
+
+{: .file }
+/etc/tinc/mynetwork/tinc.conf
+:   ~~~ conf
+    Name=mylinode1
+    Device=/dev/net/tun
+    ConnectTo=mylinode2
+    ConnectTo=mylinode3
+    ~~~
+
+8.  Create public and private keys for each tinc node.
+
+        sudo tincd -K -n mynetwork
+
+You may leave the key size value at default or increase it to 2048 or 4096 for more security.
```

Comment:
You should specify a) how to do this, and b) do so before the command, so people don't read this after already running it.